lib/adb.js

```diff
@@ -50,6 +50,28 @@ class ADB {
     }
     this.executable.defaultArgs.push('-P', this.adbPort);
   }
+
+  /**
+   * Create a new instance of `ADB` that inherits configuration from this `ADB` instance.
+   * This avoids the need to call `ADB.createADB()` multiple times.
+   * @param {object} opts - Additional options mapping to pass to the `ADB` constructor.
+   * @returns {ADB} The resulting class instance.
+   */
+  clone (opts = {}) {
+    const cloneOptions = _.defaultsDeep(opts, _.cloneDeep(this));
```

Comment:
I'm not quite sure if it is a good idea to deep clone the whole instance in case we only need particular options. Would it be possible to only pick what's actually needed? Perhaps via listing the keys of DEFAULT_OPTS objects?

Reply:
Updated to use `_.pick`. I kept `_.cloneDeep` as well so the deep options are cloned, although the shallow clone is redundant.